vnext/Microsoft.ReactNative/Utils/BatchingEventEmitter.cpp

```diff
@@ -13,18 +13,30 @@ BatchingEventEmitter::BatchingEventEmitter(Mso::CntPtr<const Mso::React::IReactC
   m_uiDispatcher = m_context->Properties().Get(ReactDispatcherHelper::UIDispatcherProperty()).as<IReactDispatcher>();
 }
 
-void BatchingEventEmitter::EmitJSEvent(int64_t tag, winrt::hstring &&eventName, JSValue &&eventObject) noexcept {
-  return EmitJSEvent(L"receiveEvent", tag, std::move(eventName), std::move(eventObject));
+void BatchingEventEmitter::DispatchEvent(
+    int64_t tag,
+    winrt::hstring &&eventName,
+    const JSValueArgWriter &eventDataWriter) noexcept {
+  return EmitJSEvent(
+      L"RCTEventEmitter",
+      L"receiveEvent",
+      [tag, eventName = std::move(eventName), &eventDataWriter](const IJSValueWriter &paramsWriter) {
```

Comment:
>&eventDataWriter [](http://example.com/codeflow?start=46&length=16)

It must be captured by value - capturing by reference may cause referencing a deleted value.

Reply:
The JSValueArgWriter is a delegate which internally is a smart pointer to an IUnknown derived implementation. Copying it just increments the ref count.

Reply:
On the second thought - it must be not an issue because the code is invoked synchronously.